components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManager.java

```diff
@@ -0,0 +1,194 @@
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.identity.configuration.mgt.core;
+
+import org.wso2.carbon.identity.configuration.mgt.core.exception.ConfigurationManagementException;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Attribute;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resource;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceAdd;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceType;
+import org.wso2.carbon.identity.configuration.mgt.core.model.ResourceTypeAdd;
+import org.wso2.carbon.identity.configuration.mgt.core.model.Resources;
+import org.wso2.carbon.identity.configuration.mgt.core.search.Condition;
+
+/**
+ * Resource manager service interface.
+ */
+public interface ConfigurationManager {
+
+    /**
+     * This API is used to get resources from all the tenants filtered with the {@link Condition}.
+     *
+     * @param searchCondition {@link Condition} representing a search filter for resources.
+     * @return {@link Resources} object with a collection of resources matching to the given {@link Condition}.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    Resources getTenantResources(Condition searchCondition) throws ConfigurationManagementException;
+
+    /**
+     * This API is used to store a new {@link ResourceType}.
+     *
+     * @param resourceTypeAdd {@link ResourceType} create request.
+     * @return 201 created. Returns created {@link ResourceType}.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    ResourceType addResourceType(ResourceTypeAdd resourceTypeAdd) throws ConfigurationManagementException;
+
+    /**
+     * This API is used to replace an existing {@link ResourceType} with the given one or create if not.
+     *
+     * @param resourceTypeAdd Request to create the {@link ResourceType}.
+     * @return 200 ok. Returns replaced {@link ResourceType}.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    ResourceType replaceResourceType(ResourceTypeAdd resourceTypeAdd) throws ConfigurationManagementException;
+
+    /**
+     * This API is used to retrieve the {@link ResourceType}.
+     *
+     * @param resourceTypeName Name of the {@link ResourceType}.
+     * @return 200 ok. Returns the corresponding {@link ResourceType}.
+     * @throws ConfigurationManagementException Configuration Management Exception.
+     */
+    ResourceType getResourceType(String resourceTypeName)
```

Comment:
formatting issue. we can inline line 67 with 68

Reply:
Fixed in d0a5b5ad8ff1bdab587b5bd0c8c0cc2c3b6d4afb